users.js

```diff
@@ -640,6 +677,27 @@ class User {
 		}
 		return true;
 	}
+	lockName() {
+		let userid = this.userid;
+		for (let ip in this.ips) {
+			nameLockedIps[ip] = userid;
+		}
+		if (this.autoconfirmed) nameLockedUsers[this.autoconfirmed] = userid;
+		nameLockedUsers[this.userid] = userid;
+		this.namelocked = userid;
+		this.forceRename(this.genericName(), this.registered);
+		this.updateIdentity();
+
+		return true;
+	}
+	genericName() {
```

Comment:
I'm stupid and didn't notice the `this.guestNum` usage. Pretend I didn't say anything